From 73fe39bf0940d788932ef6d346bb66298eeeb5d7 Mon Sep 17 00:00:00 2001
From: Tomi Valkeinen <tomi.valkeinen@ti.com>
Date: Wed, 2 Mar 2011 12:48:35 +0200
Subject: [PATCH 072/108] OMAP: DSS2: Remove unneeded cpu_is_xxx checks

commit 0a5835185420160f4fe8a6e1d1e42cd7efa07dc7 upstream

cpu_is_omapxxx() was used previously to select the supported interfaces.
Now that the interfaces are platform devices, we no longer need to do
the check when registering the driver. Thus we can just remove the
checks.

Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
Integrated-by: Jingdong Lu <jingdong.lu@windriver.com>
---
 drivers/video/omap2/dss/core.c |   18 ++++++------------
 1 files changed, 6 insertions(+), 12 deletions(-)

diff --git a/drivers/video/omap2/dss/core.c b/drivers/video/omap2/dss/core.c
index 8cfedae..c2f930b 100644
--- a/drivers/video/omap2/dss/core.c
+++ b/drivers/video/omap2/dss/core.c
@@ -203,12 +203,10 @@ static int omap_dss_probe(struct platform_device *pdev)
 		goto err_venc;
 	}
 
-	if (cpu_is_omap34xx()) {
-		r = dsi_init_platform_driver();
-		if (r) {
-			DSSERR("Failed to initialize DSI platform driver\n");
-			goto err_dsi;
-		}
+	r = dsi_init_platform_driver();
+	if (r) {
+		DSSERR("Failed to initialize DSI platform driver\n");
+		goto err_dsi;
 	}
 
 	r = dss_initialize_debugfs();
@@ -240,8 +238,7 @@ static int omap_dss_probe(struct platform_device *pdev)
 err_register:
 	dss_uninitialize_debugfs();
 err_debugfs:
-	if (cpu_is_omap34xx())
-		dsi_uninit_platform_driver();
+	dsi_uninit_platform_driver();
 err_dsi:
 	venc_uninit_platform_driver();
 err_venc:
@@ -265,10 +262,7 @@ static int omap_dss_remove(struct platform_device *pdev)
 	venc_uninit_platform_driver();
 	dispc_uninit_platform_driver();
 	rfbi_uninit_platform_driver();
-	if (cpu_is_omap34xx()) {
-		dsi_uninit_platform_driver();
-	}
-
+	dsi_uninit_platform_driver();
 	dss_uninit_platform_driver();
 
 	dss_uninit_overlays(pdev);
-- 
1.7.4.1

